src/python/foglamp/core/audit_trail_db_services.py

```diff
@@ -24,27 +24,39 @@ class _Severity(IntEnum):
     WARNING = 3
     INFORMATION = 4
 
-async def read_audit_entries(limit=None):
+async def read_audit_entries(limit=None, offset=None, source=None, severity=None):
     """
 
     Args:
         limit: the number of audit entries returned to the number specified
+        source: filter the audit entries to be only those from the specified source
+        severity: filter the audit entries to only those of the specified severity
+        offset: skip the first n entries in the audit table, used with limit to implemented paged interfaces
 
     Returns:
         list of audit trail entries sorted with most recent first
     """
     conn = await asyncpg.connect(database=__DB_NAME)
-    _limit_clause = " LIMIT $1" if limit else " "
+    
+    _limit_clause = " LIMIT {0}".format(limit) if limit else " "
+    _offset_clause = " "
+    if limit:
+        _offset_clause = " OFFSET {0}".format(offset) if offset else " "
+
+    _where_clause = " "
```

Comment:
I'd rather we built where clauses based on individual items so we don't have to enumerate all of the combinations. 

You can use
   _where_clause = " WHERE 1 = 1 "
  if source:
    _where_clause = _where_clause + "AND source = {0}'.format(source)
 if severity:
    _where_clause = _where_clause + "AND severity = {0}'.format(_Severity[severity].value)

This way when we can more in the future we do not get an exponential explosion of if statements